trunk/source/PEAtoBoogie/src/de/uni_freiburg/informatik/ultimate/pea2boogie/req2pea/ReqCheckAnnotator.java

```diff
@@ -487,20 +496,32 @@ private Statement genAssertNonVacuous(final PatternType<?> req, final PhaseEvent
 
 		// check that one of those phases is eventually reached.
 		final List<Expression> checkReached = new ArrayList<>();
+		final Map<Phase, CDD> enforcedInv = new HashMap<>();
+
 		if (pnr > 0) {
 			for (int i = 0; i < phases.size(); i++) {
 				final PhaseBits bits = phases.get(i).getPhaseBits();
 				if (bits == null || (bits.getActive() & 1 << pnr - 1) == 0) {
 					checkReached.add(genComparePhaseCounter(i, mSymbolTable.getPcName(aut), bl));
 				}
+				// Store state invariants of PEA locations where max DC phase is not active.
+				if (bits.getActive() < (1 << pnr - 1)) {
```

Comment:
If the mask `1 << pnr - 1` is now used multiple times, perhaps give it its own variable and a nice name. I am struggling a bit to understand what exactly the condition here is -- its weird for me to see a mask used as mask and as comparison.